### ISSUE TITLE
Allow `source` to not be a Promise

### ISSUE THREAD

**Issue by sindresorhus** [mokkabonna/inquirer-autocomplete-prompt#22]
You might already have gotten the data before the prompt is created. Instead of forcing me to wrap my results in a promise, it would be nice if you just did: `Promise.resolve().then(() => opts.source)`.

*Note, I'm intentionally not suggesting `Promise.resolve(opts.source())` as that would not catch any synchronous error.*

Right now, I'm doing:

```js
source: (answers, input) => Promise.resolve().then(() => filterProcesses(input, processes))
```

**Comment by mokkabonna** [comment]
Added a while ago.

**Comment by mokkabonna** [comment]
Though I had at least. Added it now.

But I had to use a slightly different appraoch to not mess to much with test or change sync behaviour. With your example the souce call is async, with the current code it is not.

So calls it sync like before but wraps it in a promise after the call, and catches errors.

https://github.com/mokkabonna/inquirer-autocomplete-prompt/blob/0d3b41c727ad0c87586a7e540816a3c97dd2730c/index.js#L205-L211
